pyomo/contrib/viewer/ui.py

```diff
@@ -23,46 +23,48 @@
 """
 A simple GUI viewer/editor for Pyomo models.
 """
-from __future__ import division, print_function, absolute_import
-
 __author__ = "John Eslick"
 
 import os
 import logging
+
 try:
     from IPython import get_ipython
 except ImportError:
+
     def get_ipython():
         raise AttributeError("IPython not available")
-import pyomo.contrib.viewer.report as rpt
-import pyomo.environ as pyo
 
-_log = logging.getLogger(__name__)
 
+import pyomo.contrib.viewer.report as rpt
+import pyomo.environ as pyo
 from pyomo.contrib.viewer.qt import *
 from pyomo.contrib.viewer.model_browser import ModelBrowser
 from pyomo.contrib.viewer.residual_table import ResidualTable
 from pyomo.contrib.viewer.model_select import ModelSelect
 from pyomo.contrib.viewer.ui_data import UIData
+from pyomo.common.fileutils import this_file_dir
 
-_mypath = os.path.dirname(__file__)
+_log = logging.getLogger(__name__)
+
+_mypath = this_file_dir()
 try:
-    _MainWindowUI, _MainWindow = \
-        uic.loadUiType(os.path.join(_mypath, "main.ui"))
+    _MainWindowUI, _MainWindow = uic.loadUiType(os.path.join(_mypath, "main.ui"))
 except:
     _log.exception("Failed to load UI files.")
     # This lets the file still be imported, but you won't be able to use it
     # Allowing this to be imported will let some basic tests pass without PyQt
     class _MainWindowUI(object):
         pass
+
     class _MainWindow(object):
         pass
 
+
 if not qt_available:
-    for _err in qt_import_errors:
-        _log.error(_err)
```

Comment:
More changes from https://github.com/Pyomo/pyomo/pull/1066 that I think we want to keep.